src/counterexamples/linear_order_with_pos_mul_pos_eq_zero.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2021 Johan Commelin (inspired by Kevin Buzzard, copied by Damiano Testa).
+All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin (inspired by Kevin Buzzard, copied by Damiano Testa)
+-/
+import algebra.ordered_monoid
+
+/-!
+An example of a `linear_ordered_comm_monoid_with_zero` in which the product of two positive
+elements vanishes.
+
+This is the monoid with 3 elements `0, ε, 1` where `ε ^ 2 = 0` and everything else is forced.
+The order is `0 < ε < 1`.  Since `ε ^ 2 = 0`, the product of strictly positive elements can vanish.
+
+Relevant Zulip chat:
+https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/mul_pos
```

Comment:
It would be cool to provide the exact statement of the proposed lemma, but prefixed with a `\not`